pkg/controller/migcluster/validation.go

```diff
@@ -346,3 +356,46 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	clusterClient, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	clusterOperatorVersion, err := cluster.GetOperatorVersion(clusterClient)
+
+	if err != nil {
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     ClusterOperatorVersionNotFound,
+			Status:   True,
+			Reason:   VersionNotFound,
+			Category: Critical,
+			Message:  fmt.Sprintf("The configmap key could not be found. See logs for details."),
+		})
+
+		return liberr.Wrap(err)
+	}
+
+	hostOperatorVersion, err := cluster.GetOperatorVersion(r)
+	operatorVersionMatchesHost := clusterOperatorVersion == hostOperatorVersion
+
+	if !operatorVersionMatchesHost {
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     OperatorVersionMismatch,
+			Status:   True,
+			Reason:   VersionCheckFailed,
+			Category: Critical,
```

Comment:
Another place we need to switch to warn for the upcoming release instead of block.